### PR TITLE
feat: SuperchainWETH redesign

### PR DESCRIPTION
**Pull request by agusduha**
Closes OPT-264

### CODE REVIEW

File: packages/contracts-bedrock/src/L2/SuperchainWETH.sol [hunk 1]
```diff
@@ -37,63 +43,48 @@ contract SuperchainWETH is WETH98, ISuperchainWETH, ISemver {
...
-            IETHLiquidity(Predeploys.ETH_LIQUIDITY).mint(wad);
+            IETHLiquidity(Predeploys.ETH_LIQUIDITY).mint(_amount);
         }
 
-        // Get source chain ID.
-        uint256 source = messenger.crossDomainMessageSource();
-
-        // Mint to user's balance.
-        _mint(dst, wad);
+        _mint(_to, _amount);
```
Comment (0xDiscotech) [inline packages/contracts-bedrock/src/L2/SuperchainWETH.sol]: This should go before the call to follow CEI, right?

Comment (0xDiscotech) [reply packages/contracts-bedrock/src/L2/SuperchainWETH.sol]: I know this is copy-pasted logic, but to me would be better to follow the pattern if it doesn't harm

File: packages/contracts-bedrock/src/L2/SuperchainWETH.sol [hunk 2]
```diff
@@ -9,18 +9,24 @@ import { Predeploys } from "src/libraries/Predeploys.sol";
...
+contract SuperchainWETH is WETH98, ICrosschainERC20, ISemver {
+    /// @notice A modifier that only allows the SuperchainTokenBridge to call
+    modifier onlySuperchainTokenBridge() {
+        if (msg.sender != Predeploys.SUPERCHAIN_TOKEN_BRIDGE) revert Unauthorized();
+        _;
+    }
+
     /// @notice Semantic version.
     /// @custom:semver 1.0.0-beta.6
     string public constant version = "1.0.0-beta.6";
```
Comment (0xDiscotech) [inline packages/contracts-bedrock/src/L2/SuperchainWETH.sol]: Missing change over version

File: packages/contracts-bedrock/test/L2/SuperchainWETH.t.sol [hunk 3]
```diff
@@ -121,237 +130,189 @@ contract SuperchainWETH_Test is CommonTest {
...
-    /// @notice Tests that the sendERC20 function can be called with a sufficient balance on a
-    ///         custom gas token chain. Also tests that the proper calls are made and the proper
-    ///         events are emitted but ETH is not burned via the ETHLiquidity contract.
-    /// @param _amount The amount of WETH to send.
-    /// @param _chainId The chain ID to send the WETH to.
-    function testFuzz_sendERC20_sufficientFromCustomGasTokenChain_succeeds(uint256 _amount, uint256 _chainId) public {
-        // Assume
-        vm.assume(_chainId != block.chainid);
+    /// @notice Tests the `crosschainMint` with custom gas token succeeds and emits the `CrosschainMinted` event.
+    function testFuzz_crosschainMint_fromBridgeCustomGasTokenChain_succeeds(address _to, uint256 _amount) public {
```
Comment (0xDiscotech) [inline packages/contracts-bedrock/test/L2/SuperchainWETH.t.sol:182]: I think it'd be a good idea to use Foundry's `function expectCall(address callee, bytes calldata data, uint64 count) external;` vm cheatcode setting `count` to zero `mint` call over ETHLiquidity - so we ensure that's not called.

The same could be applied for `crosschainBurn()`

**Review by 0xDiscotech — COMMENTED** [review]
Looking good ser! 
Left few small comments in addittion that on the tests, the assertions after a revert could be removed, IMO no need to add them after `expectRevert` was used - not a big deal tho

**Review by 0xDiscotech — DISMISSED** [review]

**Comment by agusduha** [comment]
Closing, follows in https://github.com/defi-wonderland/optimism/pull/101

**Pull request closed without merging**